ambry-replication/src/main/java/com.github.ambry.replication/CloudToStoreReplicationManager.java

```diff
@@ -334,6 +374,10 @@ public void onPartitionBecomeStandbyFromBootstrap(String partitionName) {
     public void onPartitionBecomeLeaderFromStandby(String partitionName) {
       logger.info("Partition state change notification from Standby to Leader received for partition {}",
           partitionName);
+      if (!partitionsToReplicate.isEmpty() && !partitionsToReplicate.contains(partitionName)) {
```

Comment:
Since you do this multiple times, suggest utility like:
boolean shouldReplicatePartition(String partitionName)
which does the check and the logging.